# MAP433 - EN2

## Importation des librairies

In [1]:
import numpy as np
import matplotlib.pyplot as plt
import numpy.random as npr
import scipy.stats as sps
import pandas as pd

## 1 - EMV de $\theta = (\mu_0,\mu_1,\sigma_0^2,\sigma_1^2)$

In [2]:
x = [13, 130, 39, 33,10,13,68,18,3,11,38,23,60,5,9,59,5,86,22,70,58,3,167,15,30,8,20,67,26,19]
y = [14, 8, 20, 3, 138, 122, 78, 69, 111, 3, 128, 31, 18, 35, 111, 109, 36, 27, 32, 35, 12, 27, 8, 3, 80, 91, 68, 66, 176, 15]
X = np.array(x)
Y = np.array(y)

In [3]:
mu0 = np.sum(X)/X.size
mu1 = np.sum(Y)/Y.size
sigma0 = (np.linalg.norm(X - mu0*np.ones(X.size))**2)/X.size
sigma1 = (np.linalg.norm(Y - mu1*np.ones(Y.size))**2)/Y.size
print("mu0 = ", mu0, "\n")
print("mu1 = ", mu1, "\n")
print("sigma0^2 = ", sigma0, "\n")
print("sigma1^2 = ", sigma1, "\n")

mu0 =  37.6 

mu1 =  55.8 

sigma0^2 =  1431.8400000000004 

sigma1^2 =  2251.36 



## 2 - Test du rapport de vraisemblance généralisé de niveau $\alpha$ de $H_0 : \sigma_0^2 = \sigma_1^2 \quad contre \quad H_1 :  \sigma_0^2 > \sigma_1^2$

On a : $R_\alpha = \Lambda^{'}(Z) > q_{F(n-1,p-1)^{1-\alpha}}$

## 3 - Calcul de la p-valeur du test

In [4]:
val = ((Y.size-1)*np.linalg.norm(X - mu0*np.ones(X.size))**2)/((X.size -1)*np.linalg.norm(Y - mu1*np.ones(Y.size))**2)
alpha_chap = 1 - sps.f.cdf(val, X.size -1, Y.size -1)
print(alpha_chap)

0.8855279309812643


Interprétation : $\alpha_{chap} \approx 0.88$ est la p-valeur du test, c-à-d la valeur en deça de laquelle on cesse de commettre l'erreur de type 1: Rejeter H0 à tort.
Comme ici, $\alpha = 0.01 < \alpha_{chap} \approx 0.88$, on ne rejètera pas H0 alors que la probabilité d'observer $\sigma_0 = \sigma_1$ est de moins de $0.01$ (Sous reserve que le modèle est correct).
<br/> <b>Conclusion:</b> Comme $\alpha_{chap}$ est grand, alors, soit le modèle gaussien considéré avec deux variances differentes n'est pas correct, soit l'hypothèse nulle n'est pas "bonne" (le test n'est pas concluant pour cette hypothèse).

## 5 - Test de niveau $\alpha = 0.01$ de l'hypothese $\mu_0 = \mu_1 \quad contre \quad \mu_0 \neq \mu_1$

Le test est defini par la region de rejet $R_\alpha = \{ Z \in \mathbb{R}^{n+p}, \sqrt{\frac{np}{n+p}}\frac{|\hat{\mu_0} - \hat{\mu_1}|}{S_{n,p}(Z)} > t\}$ avec t = 

In [5]:
alpha = 0.01
t = sps.t.ppf(1 - alpha/2, X.size+Y.size-2)
print(t)

2.6632869538098674


## 6 - p-valeur du test

In [6]:
SnpCarre = (np.linalg.norm(X - mu0*np.ones(X.size))**2 + np.linalg.norm(Y - mu1*np.ones(Y.size))**2)/(X.size+Y.size-2)
print(SnpCarre)
alpha_chap2 = 2*(1 - sps.t.cdf(np.sqrt((X.size*Y.size)/(X.size+Y.size))*np.abs(mu0 - mu1)/np.sqrt(SnpCarre), X.size+Y.size-2))
print(alpha_chap2)

1905.1034482758623
0.11175016168340379


Interprétation : $\alpha_{chap} \approx 0.11$ est la p-valeur du test, c-à-d la valeur en deça de laquelle on cesse de commettre l'erreur de type 1: Rejeter H0 à tort.
Comme ici, $\alpha = 0.01 < \alpha_{chap} \approx 0.11$, on ne rejètera pas H0 et la probabilité d'observer $\mu_0 = \mu_1$ est de moins de $0.01$ (Sous reserve que le modèle est correct).
<br/> <b>Conclusion:</b> Comme $\alpha_{chap2}$ est petit, le test est efficace pour cette hypothèse nulle.

## 7- Test de niveau $\alpha$ avec $\alpha = 0.01$ et $H_0 : \mu_0 \leq \mu_1 \quad contre \quad \mu_0 > \mu_1$

Le test est defini par la region de rejet: $R_\alpha = \{ Z \in \mathbb{R}^{n+p},\sqrt{\frac{np}{n+p}}\frac{\hat{\mu_0} - \hat{\mu_1}}{S_{n,p}(Z)} > t\}$ avec t = 

In [7]:
alpha = 0.01
t = sps.t.ppf(1-alpha, X.size+Y.size-2)
print(t)

2.3923774702828906


## 8- p-valeur du test

In [8]:
alpha_chap3 = (1 - sps.t.cdf(np.sqrt((X.size*Y.size)/(X.size+Y.size))*(mu0 - mu1)/np.sqrt(SnpCarre), X.size+Y.size-2))
print(alpha_chap3)

0.9441249191582981


Interprétation : $\alpha_{chap} \approx 0.94$ est la p-valeur du test, c-à-d la valeur en deça de laquelle on cesse de commettre l'erreur de type 1: Rejeter H0 à tort.
Comme ici, $\alpha = 0.01 < \alpha_{chap} \approx 0.94$, on ne rejètera pas H0 alors que la probabilité d'observer $\mu_0 \leq  \mu_1$ est de moins de $0.01$ (Sous reserve que le modèle est correct).
<br/> <b>Conclusion:</b> Comme $\alpha_{chap3}$ est grand, alors, soit le modèle gaussien considéré avec une même variance n'est pas correct, soit l'hypothèse nulle n'est pas "bonne" (le test n'est pas concluant pour cette hypothèse).